src/components/Router/index.js

```diff
@@ -291,13 +291,44 @@ class HistoryRouter extends Component {
     this.props.options.events.emit('complete', data)
   }
 
-  triggerOnError = (apiResponse) => {
-    const { status, response } = apiResponse
-    if (status === 0) return
+  formattedError = (response, status) => {
+    if (typeof response === 'string') {
+      // TODO: this if statement should be deleted once all APIs start using the same signature for responses
+      // Currently detect_documents returns just a string. Examples:
+      // `Could not decode token: hello`
+      // `Token has expired.`
+      // Telephony returns the a JSON response
+      // {“unauthorized”:”Could not decode token: hello"}
+      // {"unauthorized":"Token has expired."}
+      // Tickets in backlog to update all APIs to use signature similar to main Onfido API
+      let message;
+      try {
+        const jsonRes = JSON.parse(response)
+        message = jsonRes.unauthorized
```

Comment:
Are we guaranteed to always get a JSON with `unauthorized` object key?

Reply:
No, only for one service and for authorization errors. I added `jsonRes.unauthorized || response` to handle all other errors that we might not be aware of.